### PR TITLE
LPS-170159 Replace PUT request with PATCH request

### PR DESCRIPTION
**Pull request by devpedrotavares**
[LPS-170159](https://issues.liferay.com/browse/LPS-170159)

The bug is happening because the client in making a PUT request passing only partial data on the object entry (because each tab has its own form element). Since the PUT endpoint requires the whole resource (at least with all mandatory fields), the request fails.
The proposed solution is to make a PATCH request instead, since it supports partial data from the client on the resource to be updated.

### CODE REVIEW

**Comment by devpedrotavares** [comment]
In my understanding, the ideal solution would be to create an endpoint `patchByExternalReferenceCode` and call it. However, when I tried to do this, restBuilder did not seem to work properly (image below) as the generate patch method calls `getObjectEntryByExternalReferenceCode` instead of  the existing `getByExternalReferenceCode`.
So I opted for calling the existing patch endpoint.

invalid endpoint

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/object_entry/form.jsp [hunk 1]
```diff
@@ -86,14 +86,13 @@ portletDisplay.setURLBack(backURL);
...
 			const postPath = scope === 'site' ? pathScopedBySite : contextPath;
 
-			let putPath = scope === 'site' ? pathScopedBySite : contextPath;
+			let patchPath = scope === 'site' ? pathScopedBySite : contextPath;
 
-			putPath = putPath.concat(
-				'/by-external-reference-code/',
-				`\${externalReferenceCode}`
+			patchPath = patchPath.concat(
+				'/<%= (objectEntry == null) ? "" : objectEntry.getId() %>'
```
Comment (carolmariaabb) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/object_entry/form.jsp:92]: Hi @devpedrotavares,

We must continue to use the `External Reference Code` to save the entry.. 

That said, or you can try to add the `patch` [here](https://github.com/liferay/liferay-portal/blob/7b11a1fa4186195fe52164a783d3f87bca7fa98b/modules/apps/object/object-rest-impl/rest-openapi.yaml#L189-L247) and run `gw buildRest`, or you can try a different approach to fix this issue.

Please let me know if you need any assistance with this.

Thanks!

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by carolmariaabb** [comment]
> In my understanding, the ideal solution would be to create an endpoint `patchByExternalReferenceCode` and call it. However, when I tried to do this, restBuilder did not seem to work properly (image below) as the generate patch method calls `getObjectEntryByExternalReferenceCode` instead of the existing `getByExternalReferenceCode`. So I opted for calling the existing patch endpoint.
> 
> invalid endpoint

Could you send a message in the #help-objects slack channel so we can ask the Headless Team for guidance? ? Thanks!

**Comment by carolmariaabb** [comment]
As discussed with the team, the best approach to fix this bug is to change the behavior of the PUT (ObjectEntryResourceImpl -> putByExternalReferenceCode,) as follows: 

- Merge the existing values ​​in the database with the new values, prioritizing the new values. 
- If the user does not send a certain value, we will consider the one already saved in the database.

Therefore, I'll close this PR.

/cc @devpedrotavares

**Pull request closed without merging**